### DISCUSSION TITLE
Unable to get gitea-pull-request mode working

### DISCUSSION THREAD

**Discussion by brianread108** [WeblateOrg/weblate#12024] [Q&A]
I am not seeing the pull request being created when I commit a change. I've tried various options in the "Repository push URL" parameter. If I fill it in with an http with username and token then I get:"Option is not available for linked repositories. Setting from linked component will be used." if I leave it blank then no pull request is created. The linked gitea repo is sucessfully accessed to create a translation component.  I have the gitea username and token configured in ENV variables.  I am using the docker image.

**Comment by nijel** [comment]
Configure that on the main component in Weblate, not on linked one, see https://docs.weblate.org/en/latest/vcs.html#weblate-internal-urls

**Reply by brianread108** [reply]
> Configure that on the main component in Weblate, not on linked one, see https://docs.weblate.org/en/latest/vcs.html#weblate-internal-urls

All my components (bar one) where created using the Add Components Addon. So I guess they are all "linked" to the one that was created specifically? In that case do I put the http with the username and token in that one and then the other linked ones should have the  "weblate://project/component" in the Push URL or should it stay blank?

**Reply by nijel** [reply]
It should stay blank, these use settings from the main component as they share the underlying Git repository.

**Comment by brianread108** [comment]
However I am still not getting the pull request through to the gitea system. Even when it is blank.  I am not now getting the Alert to say that the request has failed. I also tried the master component, and the same occurred, no pull request, but no alert.

**Comment by brianread108** [comment]
I am seting the GITEA credentials in the compose parameters: 

      WEBLATE_GITEA_USERNAME: brianr
      WEBLATE_GITEA_TOKEN: xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
      WEBLATE_GITEA_HOST: gitea.bjsystems.co.uk

As well as in the "Repository push URL" property:

http://brianr:xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx@gitea.bjsystems.co.uk/brianr/smeserver-manager-locale (I've tried wit .git and without)

Maybe its because it is only "http"? I'm totally internal at the moment.

**Reply by nijel** [reply]
What error do you get? The scheme should be parsed from the repository URL, but in case it doesn't work, it can be configured, see https://docs.weblate.org/en/latest/admin/config.html#configuring-version-control-credentials. Unfortunately this is not exposed to Docker environment variables.

**Comment by brianread108** [comment]
I am not seeing an error.  At least no alert is showing. It just says the commit was done (sucessfully?)

Can I present the "scheme" parameter using a python structure to the docker version?

and what are the possible value for scheme?

**Comment by brianread108** [comment]
Just found this is the RSS feed, not very helpful!:

`Repository push failed at 2024-07-10 09:51:43.087803+00:00 on None by None`

**Reply by nijel** [reply]
More detailed will be visible in history. For push failure, you should also get an alert with the details.

**Comment by brianread108** [comment]
I've switched to my project's remote gitea, which does have https, but no ssh access. I now get this on the push:
`Could not push SMESERVER SM2 Translate Remote/Backup: No RSA host key is known for [xx.yyyyy.org]:2222 and you have requested strict checking. Host key verification failed. fatal: Could not read from remote repository. Please make sure you have the correct access rights and the repository exists. (128)
`
Does it need ssh access as well as through the api?

I have read sucessfully from the gitea repo to setup the component.

**Reply by nijel** [reply]
SSH might be used for pushing the repository.
